jl4-core/src/L4/Desugar.hs

```diff
@@ -0,0 +1,162 @@
+  Gt         ann e1 e2 -> Gt        ann (carameliseExpr e1) (carameliseExpr e2)
+  Proj       ann e n   -> Proj ann (carameliseExpr e) n
+  Var        ann n     -> Var  ann n
+  Lam        ann sig e -> Lam ann sig (carameliseExpr e)
+  App        ann n es  -> App ann n (fmap carameliseExpr es)
+  AppNamed   ann n nes morder -> AppNamed ann n (fmap caramliseNamedExpr nes) morder
+  IfThenElse ann b t e -> IfThenElse ann (carameliseExpr b) (carameliseExpr t) (carameliseExpr e)
+  Regulative ann o -> Regulative ann (carameliseObligation o)
+  Consider   ann e branches -> Consider ann (carameliseExpr e) (fmap carameliseBranch branches)
+  Lit        ann l -> Lit ann l
+  Percent    ann e -> Percent ann (carameliseExpr e)
+  List       ann es -> List ann (fmap carameliseExpr es)
+  Where      ann e ds -> Where ann (carameliseExpr e) (fmap carameliseLocalDecl ds)
+  Event      ann ev -> Event ann (carameliseEvent ev)
+
+carameliseLocalDecl :: HasName n => LocalDecl n -> LocalDecl n
+carameliseLocalDecl = \ case
+  LocalDecide ann decide -> LocalDecide ann (carameliseDecide decide)
+  LocalAssume ann assume -> LocalAssume ann assume
+
+carameliseDecide :: HasName n => Decide n -> Decide n
+carameliseDecide = \ case
+  MkDecide ann tySig appForm expr ->
+    MkDecide ann tySig appForm (carameliseExpr expr)
+
+carameliseBranch :: HasName n => Branch n -> Branch n
+carameliseBranch = \ case
+  When ann pat e -> When ann pat (carameliseExpr e)
+  Otherwise ann e -> Otherwise ann (carameliseExpr e)
+
+carameliseEvent :: HasName n => Event n -> Event n
+carameliseEvent = \ case
+  MkEvent { anno, party, action, timestamp, atFirst} ->
+    MkEvent
+      { anno
+      , party = carameliseExpr party
+      , action = carameliseExpr action
+      , timestamp = carameliseExpr timestamp
+      , atFirst
+      }
+
+carameliseObligation :: HasName n => Obligation n -> Obligation n
+carameliseObligation = \ case
+  MkObligation { anno, party, action, due, hence, lest} ->
+    MkObligation
+      { anno
+      , party = carameliseExpr party
+      , action = carameliseRAction action
+      , due = fmap carameliseExpr due
+      , hence = fmap carameliseExpr hence
+      , lest = fmap carameliseExpr lest
+      }
+
+carameliseRAction :: HasName n => RAction n -> RAction n
+carameliseRAction = \ case
+  MkAction { anno, action, provided } ->
+    MkAction
+      { anno
+      , action
+      , provided = fmap carameliseExpr provided
+      }
+
+caramliseNamedExpr :: HasName n => NamedExpr n -> NamedExpr n
+caramliseNamedExpr = \ case
+  MkNamedExpr ann n e -> MkNamedExpr ann n (carameliseExpr e)
+
+-- | We desugar expressions during typechecking, for example, @2 PLUS 3@ is
+-- turned into the function @\_\_PLUS\_\_ 2 3@. This is less readable during natural
+-- language generation, so we undo some of our desugaring and add the syntactic sugar
+-- back.
+--
+-- We call this process "caramelise". Primarily, because fendor likes caramel,
```

Comment:
are you talking of yourself in the third person now :D (/jk)

Reply:
Just to make the comment resistant to line changes :)

Reply:
Yeah I get it, was a joke :3